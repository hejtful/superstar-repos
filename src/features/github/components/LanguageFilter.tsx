import { ChangeEvent, EventHandler } from 'react';

import { languages } from '../constants';

import styles from './LanguageFilter.module.css';

interface Props {
  onChange: EventHandler<ChangeEvent>;
}

export function LanguageFilter({ onChange }: Props) {
  return (
    <label htmlFor="language-filter" className={styles.label}>
      Language:
      <select
        id="language-filter"
        className={styles.select}
        onChange={onChange}
      >
        <option>Any</option>
        {languages.map((language) => (
          <option value={language} key={language}>
            {language}
          </option>
        ))}
      </select>
    </label>
  );
}
