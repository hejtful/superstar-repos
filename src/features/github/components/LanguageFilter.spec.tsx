import { render } from '@testing-library/react';

import { LanguageFilter } from './LanguageFilter';

it('should render properly', () => {
  const { asFragment } = render(<LanguageFilter onChange={jest.fn()} />);

  expect(asFragment()).toMatchSnapshot();
});
