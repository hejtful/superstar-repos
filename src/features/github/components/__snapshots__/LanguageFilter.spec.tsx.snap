// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<DocumentFragment>
  <label
    class="label"
    for="language-filter"
  >
    Language:
    <select
      class="select"
      id="language-filter"
    >
      <option>
        Any
      </option>
      <option
        value="JavaScript"
      >
        JavaScript
      </option>
      <option
        value="Python"
      >
        Python
      </option>
      <option
        value="Java"
      >
        Java
      </option>
      <option
        value="PHP"
      >
        PHP
      </option>
      <option
        value="C++"
      >
        C++
      </option>
      <option
        value="C"
      >
        C
      </option>
      <option
        value="C#"
      >
        C#
      </option>
      <option
        value="Ruby"
      >
        Ruby
      </option>
      <option
        value="Scala"
      >
        Scala
      </option>
      <option
        value="Haskell"
      >
        Haskell
      </option>
      <option
        value="Shell"
      >
        Shell
      </option>
      <option
        value="Elm"
      >
        Elm
      </option>
      <option
        value="Rust"
      >
        Rust
      </option>
      <option
        value="Kotlin"
      >
        Kotlin
      </option>
    </select>
  </label>
</DocumentFragment>
`;
